File: 201912-1/python/main.py
def judge(n):
    if n % 7 == 0:
        return False
    i = 1
    while n/i >= 1:
        if int(n/i) % 10 == 7:
            return False
        i *= 10
    return True


def main():
    total = int(input())
    counts = [0, 0, 0, 0]
    i = 0
    while total > 0:
        if judge(i + 1):
            total -= 1
        else:
            counts[i % 4] += 1
        i += 1
    for count in counts:
        print(count)


if __name__ == '__main__':
    main()

File: 201912-1/cpp/main.cpp
#include <cstdio>

using namespace std;

bool isJump(int n)
{
    if (n % 7 == 0)
    {
        return false;
    }

    for (int i = 1; n / i >= 1; i *= 10)
    {
        if ((n / i) % 10 == 7)
        {
            return false;
        }
    }

    return true;
}

int main()
{
    int total;
    scanf("%d", &total);

    int counts[4] = {0, 0, 0, 0};

    for (int i = 0; total > 0; i++)
    {
        if (isJump(i + 1))
        {
            total--;
        }
        else
        {
            counts[i % 4]++;
        }
    }

    for (int i = 0; i < 4; i++)
    {
        printf("%d\n", counts[i]);
    }

    return 0;
}